### PR TITLE
bump ui_test to 0.22

### PR DESCRIPTION
**Pull request by oli-obk**

### CODE REVIEW

File: tests/compiletest.rs [hunk 1]
```diff
@@ -86,17 +85,20 @@ fn test_config(target: &str, path: &str, mode: Mode, with_dependencies: bool) ->
...
         program,
         out_dir: PathBuf::from(std::env::var_os("CARGO_TARGET_DIR").unwrap()).join("ui"),
-        edition: Some("2021".into()), // keep in sync with `./miri run`
         threads: std::env::var("MIRI_TEST_THREADS")
             .ok()
             .map(|threads| NonZeroUsize::new(threads.parse().unwrap()).unwrap()),
         ..Config::rustc(path)
     };
+    // keep in sync with `./miri run`
+    config.comment_defaults.base().edition = Some(Spanned::dummy("2021".into())).into();
```
Comment (RalfJung) [inline tests/compiletest.rs:96]: It's a bit unclear what should be kept in sync.
```suggestion
    let edition = "2021"; // keep in sync with `./miri run`
    config.comment_defaults.base().edition = Some(Spanned::dummy(edition.into())).into();
```
But also... this is quite confusing, why do we need to `Spanned` everything?

File: tests/compiletest.rs [hunk 2]
```diff
@@ -66,8 +67,6 @@ fn test_config(target: &str, path: &str, mode: Mode, with_dependencies: bool) ->
         }
     }
     program.args.push("-Zui-testing".into());
-    program.args.push("--target".into());
-    program.args.push(target.into());
```
Comment (RalfJung) [inline tests/compiletest.rs:70]: From the CI errors, removing `--target` doesn't seem to be right.

Comment (oli-obk) [reply tests/compiletest.rs:70]: yea ^^ looks like it

**Comment by RalfJung** [comment]
@oli-obk what's the plan with this PR? Is ui_test 0.23 ready to be used by Miri?

**Comment by oli-obk** [comment]
still working on this now and then, keeping it open is not useful

**Pull request closed without merging**